Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# More provinces

## Changelog:
- Added new lore provinces
- Added a lot of new provinces and baronnies with lore name or not.

# How to test:
- Open the mod, compare the map
- Check if some provinces are too small (smaller than 20x20) or too big (larger than 35x35, 40x40 maximum for rare provinces.)
- Check if provinces and borders are pretty

# Notes:
- Some province names are inspired by the name of the lore, but not entirely. For example, Gordok's Seat by Dire Maul has become Shandaral's Seat.
- Some provinces, notably in Pandaria, have lore names but translated into Chinese.
- Made by: Malicos, Zumbal, Ercarp and ElMariuso.

# Tasks:
- [x] Fix this province.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/114919091-e575ec80-9e38-11eb-9108-0e99f5e0b36b.png)

</details>

- [x] Build new holdings once a civilization reaches region in history files
- [x] Black Morass has a lot of one province counties.
- [x] Add new duchies to regions
- [x] Rename Azuremyst Isle Valaar
- [x] Rename Barrens Crossroads
- [x] Rename Ashenvale Astranaar
- [x] Rename Azshara Eldara
- [x] Rename Kalidar duchy
- [x] Fix Caer Darrow not being a vassal of Lordaeron
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/114931172-f0377e00-9e46-11eb-8756-7963c4ac5d97.png)

</details>

- [x] Make this county not 1 prov
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/114931509-6340f480-9e47-11eb-885c-8f136a7088ad.png)

</details>

- [x] Build Kalidar holdings in 605 bookmark
- [x] Check counties don't have 2 holdings of the same type having less than 3 holdings
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/114934681-41497100-9e4b-11eb-8f4c-8d383c9514be.png)

</details>

- [x] Rename Thor Modan Solstice
- [x] Fix `geographical_region.cpp:114`
- [x] Clear removed titles from the development history file. Without this, all provs except Ahn'qiraj are 0 development.
- [x] Reassign terrain in provinceDef
  - [x] Shimmering Flats should be desert.
  - [x] The Barrens should be drylands.
  - [x] Towlong Steppes, Dread Wastes and Lower Kul Lai should be steppes.
  - [x] Jade Forest shouldn't be jungles
  - [x] Expand jungles to hills and mountains. Jungles have bigger development debuff and they matter more than hills
- [x] Connect provinces through rivers and small sea straits
- [x] Move Enkilah capital to Enkilah
- [x] Check `c_lushwater` is removed and replaced
- [x] Move this county to Deathknell
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/115166289-7829bc00-a0c3-11eb-85fa-aa158dc192f9.png)

</details>

- [x] Check region errors
- [x] Correct these borders
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/115170538-f3926a00-a0d1-11eb-917e-980b74ac7e30.png)
![image](https://user-images.githubusercontent.com/46384992/115171035-2be67800-a0d3-11eb-83ec-867875fcb36a.png)

</details>

- [x] Make it tribal
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/115457202-3ae53b80-a235-11eb-8e11-48fa341af775.png)

</details>

- [x] Update #84's regions once they're merged
- [x] Merge 1 province counties
- [x] Feudal Aerial Peek
- [x] Check how Scourge faith and culture doing in Northrend
- [x] Update these provinces
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/116147035-41b9f580-a6f0-11eb-9443-ec4d067d0702.png)

</details>
